python/nav/eventengine/plugins/juniperalertcount.py

```diff
@@ -0,0 +1,113 @@
+#
+# Copyright (C) 2022 Sikt
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""juniper alert count handler plugin"""
+
+from nav.eventengine import unresolved
+from nav.eventengine.alerts import AlertGenerator
+from nav.eventengine.plugin import EventHandler
+
+
+class JuniperAlertCountHandler(EventHandler):
+    """Accepts juniperYellowAlarmState and juniperRedAlarmState events"""
+
+    handled_types = (
+        'juniperYellowAlarmState',
+        'juniperRedAlarmState',
+    )
+
+    def handle(self):
+        event = self.event
+
+        if event.state == event.STATE_STATELESS:
+            self._logger.info(
+                'Ignoring stateless %s event: %r', event.event_type, event
+            )
+            self.event.delete()
+        elif event.state == event.STATE_START:
+            return self._handle_start()
+        elif event.state == event.STATE_END:
+            return self._handle_end()
+
+    def _handle_start(self):
+        event = self.event
+        alert = AlertGenerator(event)
+
+        if alert.alert_type not in (
+            "juniperRedAlarmOn",
+            "juniperYellowAlarmOn",
+        ):
+            self._logger.info(
+                'Ignoring %s start event with alert type %s: %r',
+                event.event_type,
+                alert.alert_type,
+                event,
+            )
+            event.delete()
+            return
+
+        alert.history_vars = {"count": alert["count"]}
```

Comment:
What happens if the alert count is zero? (Will there even be something to read from snmp at all if that happens?)

Are there separate counts for red and yellow?

Reply:
Yes, if the alert count is zero the result will we zero.

And yes, there are different counts, they are saved in to separate alarms

Reply:
So is it correct to send an alarm if the count is zero? Doesn't that mean that the event is over and we should trigger a STATE_END?

Reply:
If the count is zero an end event is sent, which results in the alert history entry being resolved (aka the end time set to when the alert count zero was detected)